src/main/java/com/ericsson/ei/controller/SubscriptionControllerImpl.java

```diff
@@ -55,133 +54,132 @@
     }
 
     @Override
     @CrossOrigin
-    @ApiOperation(value = "Removes the subscription from the database")
-    public ResponseEntity<SubscriptionResponse> deleteSubscriptionById(@PathVariable String subscriptionName) {
-        SubscriptionResponse subscriptionResponse = new SubscriptionResponse();
-        LOG.info("Subscription :" + subscriptionName + " delete started");
-        if (subscriptionService.deleteSubscription(subscriptionName)) {
-            LOG.info("Subscription :" + subscriptionName + " deleted Successfully");
-            subscriptionResponse.setMsg("Deleted Successfully");
-            subscriptionResponse.setStatusCode(HttpStatus.OK.value());
-            return new ResponseEntity<>(subscriptionResponse, HttpStatus.OK);
-        } else {
-            LOG.info("Subscription :" + subscriptionName + " delete completed :: Record not found for delete");
-            subscriptionResponse.setMsg("Record not found for delete");
-            subscriptionResponse.setStatusCode(HttpStatus.BAD_REQUEST.value());
-            return new ResponseEntity<>(subscriptionResponse, HttpStatus.BAD_REQUEST);
-        }
+    @ApiOperation(value = "Removes the subscriptions from the database")
+    public ResponseEntity<List<SubscriptionResponse>> deleteSubscriptionById(@PathVariable String subscriptionName) {
+        errorMap = new HashMap<>();
+        List<String> subscriptionNames = Arrays.asList(subscriptionName.split(","));
+
+        subscriptionNames.forEach(name -> {
+            LOG.debug("Subscription delete started :: " + name);
+            if (subscriptionService.deleteSubscription(name)) {
+                LOG.debug("Subscription was deleted successfully :: " + name);
+            } else {
+                LOG.error("Subscription was not found :: " + name);
+                errorMap.put(name, "Subscription was not found");
+            }
+        });
+        return (errorMap.isEmpty()) ? new ResponseEntity<>(HttpStatus.OK) : new ResponseEntity<>(getSubscriptionResponseList(errorMap), HttpStatus.BAD_REQUEST);
     }
 
     @Override
     @CrossOrigin
-    @ApiOperation(value = "Retrieve all the subscriptions")
+    @ApiOperation(value = "Retrieves all the subscriptions")
     public ResponseEntity<List<Subscription>> getSubscriptions() {
-        LOG.info("Subscription : get all records started");
-        List<Subscription> subscriptionList = new ArrayList<>();
+        LOG.debug("Subscription get all records started");
         try {
-            subscriptionList = subscriptionService.getSubscription();
-            return new ResponseEntity<>(subscriptionList, HttpStatus.OK);
+            return new ResponseEntity<>(subscriptionService.getSubscription(), HttpStatus.OK);
         } catch (SubscriptionNotFoundException e) {
             LOG.error(e.getLocalizedMessage());
```

Comment:
Error message should be part of the ReponseEntity,,, via errorMap or something, or directly to the RepsonseEntity body.

Reply:
HTTP Status code maybe can be NOT FOUND,, or it might be ok to return OK as status code , since EI Backend might not have any registered subscriptions and list is just empty due to that. But Not Found maybe is also an proper Http Status code.

Reply:
I think that empty list and status OK is good in this case, because collection can be empty and it is not an error.

Reply:
If the Exception is Status OK,, why do with do Log.error() ,,, the message should the be log.info().

Reply:
What text message is part of the Exception. If it is not a good message,, the add some good info of what happen to the log.info() message, e.g. "No Subscription has been registered in EI Backend" , "No subscription was found" or something.

Reply:
It's custom error and in this case it returns "Empty Subscription in repository". About log level you are right. In this endpoint it should not be as error